dashboard/src/app/dashboard/[dashboardId]/(dashboard)/WeeklyHeatmapSection.tsx

```diff
@@ -0,0 +1,165 @@
+                      session_duration: 'Session Duration',
+                    }[opt.value]
+                  }
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        </div>
+      </div>
+
+      <HeatmapGrid
+        data={current?.matrix ?? []}
+        maxValue={current?.maxValue ?? 1}
+        metricLabel={selectedMetricLabel}
+        metric={selectedMetric}
+      />
+    </div>
+  );
+}
+
+type HeatmapGridProps = {
+  data: WeeklyHeatmapMatrix[];
+  maxValue: number;
+  metricLabel: string;
+  metric: HeatmapMetric;
+};
+
+function HeatmapGrid({ data, maxValue, metricLabel, metric }: HeatmapGridProps) {
+  const dayLabels = ['Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.', 'Sun.'];
+
+  const effectiveMax = Math.max(1, maxValue);
+
+  const getCellStyle = useCallback(
+    (value: number): CSSProperties => {
+      if (value <= 0) return {};
+
+      const t = Math.log1p(value) / Math.log1p(effectiveMax);
+      const eased = Math.pow(t, 0.85);
+
+      return {
+        backgroundColor: `oklch(62% 0.17 268.71 / ${0.1 + 0.9 * eased})`,
+      };
+    },
+    [effectiveMax],
+  );
+
+  return (
+    <div className='grid grid-cols-[40px_repeat(7,1fr)] gap-x-0.5 gap-y-1'>
+      <div></div>
+      {dayLabels.map((label) => (
+        <div
+          key={label}
+          className='text-muted-foreground truncate pb-1 text-center text-[10px] leading-none font-medium'
+        >
+          {label}
+        </div>
+      ))}
+
+      {Array.from({ length: 24 }).map((_, hourIndex) => (
+        <Fragment key={`hour-${hourIndex}`}>
+          <div className='text-muted-foreground flex h-2.5 items-center justify-end pr-2 text-xs leading-none'>
+            {hourIndex % 3 === 1 ? String(hourIndex).padStart(2, '0') : ''}
+          </div>
+          {Array.from({ length: 7 }).map((_, dayIndex) => {
+            const value = data[dayIndex]?.hours[hourIndex] ?? 0;
+            return (
+              <Tooltip key={`${hourIndex}-${dayIndex}`}>
+                <TooltipTrigger asChild>
+                  <div
+                    className={cn(
+                      'hover:ring-primary/60 h-2.5 w-full rounded-sm transition-colors ring-inset hover:ring-1',
+                      value <= 0 && 'bg-gray-500/10 dark:bg-gray-400/20',
+                    )}
+                    style={getCellStyle(value)}
+                    aria-label={`${dayLabels[dayIndex]} ${String(hourIndex).padStart(2, '0')}:00 value ${value}`}
+                  />
+                </TooltipTrigger>
+                <TooltipContent
+                  side='top'
+                  className='border-border bg-popover/95 text-popover-foreground rounded-lg border p-2.5 shadow-xl backdrop-blur-sm'
+                >
+                  <div>
+                    <div className='text-popover-foreground font-medium'>
+                      {`${dayLabels[dayIndex]} ${hourIndex}:00 - ${(hourIndex + 1) % 24}:00`}
```

Comment:
This one is missing the `String(hourIndex).padStart(2, '0')`.
Since it's done quite a few times, consider adding a small utils for it maybe?